open_rarity/rarity_ranker.py

```diff
@@ -20,35 +27,75 @@ def rank_tokens(*, token_id_to_scores: dict[str, float]) -> dict[str, int]:
 
         Parameters
         ----------
-        token_ids_to_score : dict[str, float]
-            A dictionary of token ids (this can be whatever id you want, as long
-            as its unique) to the rarity score.
+        collection : Collection
+            Collection object with populated tokens
+        scorer: Scorer
+            Scorer instance
 
         Returns
         -------
-        dict[str, int]
-            A dictionary of token ids to their ranks. More specificially, it is the
-            provided tokens ids to the rank it has within the set of possible
-            tokens defined by @token_id_to_scores.
+        list[TokenRarity]
+            list of TokenRarity objects with score, rank and token information
+        """
+
+        if (
+            collection is None
+            or collection.tokens is None
+            or len(collection.tokens) == 0
+        ):
+            return collection
+
+        tokens = collection.tokens
+        scores: list[float] = scorer.score_tokens(collection, tokens=tokens)
+
+        # fail ranking if dimension of scores doesn't match dimension of tokens
+        assert len(tokens) == len(scores)
+
+        token_rarity_list: list[TokenRarity] = []
+        # augment collection tokes with score information
+        for idx, token in enumerate(tokens):
+            token_rarity_list.append(
+                TokenRarity(token=token, score=scores[idx])
+            )
+
+        return RarityRanker.rank_tokens(token_rarity_list)
```

Comment:
When we use the library from the opensea side, we have a unique id associated with each token that does cannot get passed into library.

Changing the ordering of tokens makes it more complex on the client's side if they need to map their `internal_token_id` to `TokenRarity`.

For this case, I would need to make the `Token` object hashable in the library and construct a map of `Token` to `internal_token_id` to be used to get to the desired end state of a map of `internal_token_id` to `TokenRarity`.

Reply:
Discussed offline  - we can transform of the return data-structure on the client side. Follow-up on multi-chain use-case will be discussed in the other PR @grantli-os